src/vs/workbench/api/browser/mainThreadSCM.ts

```diff
@@ -620,4 +634,13 @@ export class MainThreadSCM implements MainThreadSCMShape {
 		const provider = repository.provider as MainThreadSCMProvider;
 		provider.$onDidChangeHistoryProviderCurrentHistoryItemGroup(historyItemGroup);
 	}
+
+	async provideTextContent(resource: URI): Promise<ITextModel | null> {
+		const model = this.modelService.getModel(resource);
+		if (model) {
+			return model;
+		}
+
+		return this.modelService.createModel('', this.languageService.createById('scminput'), resource);
+	}
```

Comment:
Instead of making MainThreadSCM class implement ITextModelContentProvider, just copy that old class ScmInputContentProvider and instantiate it in the ctor of MainThreadSCM. Keeps things separate.